WordPress/src/main/java/org/wordpress/android/models/MediaUploadState.java

```diff
@@ -18,4 +22,22 @@ public static MediaUploadState fromString(String strState) {
         }
         return UPLOADED;
     }
+
+    public static String getLabel(Context context, MediaUploadState state) {
+        switch (state) {
+            case QUEUED:
+                return context.getString(R.string.media_upload_state_queued);
+            case UPLOADING:
+                return context.getString(R.string.media_upload_state_uploading);
+            case DELETING:
+                return context.getString(R.string.media_upload_state_deleting);
+            case DELETED:
+                return context.getString(R.string.media_upload_state_deleted);
+            case FAILED:
+                return context.getString(R.string.media_upload_state_failed);
+            case UPLOADED:
```

Comment:
I don't think that's a good idea as AS will not warn us that there is an unhandled case in the `switch` if we add another state.